packages/docusaurus/src/server/routes.ts

```diff
@@ -123,23 +123,9 @@ ${indent(subroutesCodeStrings.join(',\n'))}
   }
 
   Object.entries(props).forEach(([propName, propValue]) => {
-    // Inspired by https://github.com/armanozak/should-quote/blob/main/packages/should-quote/src/lib/should-quote.ts
-    const shouldQuote = ((key: string) => {
-      // Pre-sanitation to prevent injection
-      if (/[.,;:}/\s]/.test(key)) {
-        return true;
-      }
-      try {
-        // If this key can be used in an expression like ({a:0}).a
-        // eslint-disable-next-line no-eval
-        eval(`({${key}:0}).${key}`);
```

Comment:
For posterity: this use of `eval()` broke our tests running in SWC due to https://github.com/swc-project/swc/issues/5068. [Repro](https://play.swc.rs/?version=1.2.241&code=H4sIAAAAAAAAA8vMLcgvKlGoVkhUqFVIK8rPVVBySMlPLi1OLC0qLdYvLcnMKVay5sqEKUuCK9ODS3IlamhacyWBiNSyxBwQDQDrG2g4VwAAAA%3D%3D&config=H4sIAAAAAAAAA0WOSwrDMAxE76J1Fm03Bd%2BhhzCuEhz8Q%2BNATfDdK4eE7Ib5PGmnFY7MTsUKWIZCS9X%2ByFBtheHEl0oTVahVZeOu2srCVRuM1%2BP51jTkDCYz2wCeKPrk5zZYLsciDNyRTUu4ml1ZMX%2B3YezHOWXqJOa0gvoNOscen7N9PPIHwCox2bwAAAA%3D)

We are moving to a pure regex solution instead.

Reply:
I'd rather split this PR in multiple smaller ones and also add some tests for such little code changes